### ISSUE TITLE
Nitpicking: htop and # of cpus online

### ISSUE THREAD

**Issue by jorgechamorro** [htop-dev/htop#1468]
Hi!

Whenever I switch off cores (e.g echo 0 > /sys/devices/system/cpu/cpu1/online ) while htop is running, htop doesn't update the # of cores displayed properly, unless I close and open it again.

I love htop!

Cheers,

**Comment by BenBE** [comment]
The number of cores is read at startup, but cores that go offline while htop is running should be indicated accordingly.

Thus the question: If you start htop with all CPUs online and put one core offline while htop is running, does the CPU meter show the corresponding core as offline?

**Comment by jorgechamorro** [comment]
Say there are eight cores and eight cpu bargraphs showing in htop, and in another window I turn off a core, htop keeps displaying eight cores as if nothing had happened. Now, say I'm seeing only seven because I turned off one before launching htop, and I turn that core back on, again, htop keeps displaying only seven cores when in reality there are now eight online.

**Comment by Explorer09** [comment]
> Say there are eight cores and eight cpu bargraphs showing in htop, and in another window I turn off a core, htop keeps displaying eight cores as if nothing had happened. Now, say I'm seeing only seven because I turned off one before launching htop, and I turn that core back on, again, htop keeps displaying only seven cores when in reality there are now eight online.

Adjusting the number of bars in the display can ruin the meter layouts the user may have carefully set. I would argue that this is "not a bug".

**Comment by BenBE** [comment]
Unless hiding offline CPUs is active, all 8 cores (even if some offline) should be displayed; with some of the cores labelled as offline. If that's not the case, this is a bug and should be investigated.

If hiding of offline cores is active, then updating the layout may cause significant flicker if one or more cores are activated/deactivated in quick succession. IIRC there was some discussion about this a while back, when hiding of offline CPUs was implemented. @fasterit ???

**Comment by jorgechamorro** [comment]
Yep, no need to re-scan so often, maybe just once every n secs? That would be great.

**Comment by jorgechamorro** [comment]
And perhaps behind a flag 're-check # of cores every [n] secs' ?

**Comment by Explorer09** [comment]
@jorgechamorro Why the "re-check number of cores every n secs" option? It seems like we are tackling the wrong problem. It's unlikely that the "maximum supported CPUs" can be changed for a particular hardware, and the real option we need is "show all offline CPUs" so the meters will allocate space for CPUs that are offline at the moment.

Dynamically adding or removing meter bars can break the layout, which is something we should avoid doing.

**Comment by jorgechamorro** [comment]
You mean always show all cores even disabled ones, but the disabled ones' bargraph shows 'offline' or something? That of course would be great. But still, then, how often would you check/update the online/offline status of the cores?

**Comment by Explorer09** [comment]
@jorgechamorro When a CPU bar is created, it's online/offline status should update in the same interval as CPU percentages. Not doing so would be a bug.

**Comment by jorgechamorro** [comment]
> @jorgechamorro When a CPU bar is created, it's online/offline status should update in the same interval as CPU percentages. Not doing so would be a bug.

Ok! But that's not what's happening here. `htop --v` gives me `htop 3.0.5` maybe that's the problem? Thanks.

**Comment by BenBE** [comment]
Please retry with a newer version like `3.3.0`. There has been quite a bit of changes since `3.0.5` was released.

**Comment by jorgechamorro** [comment]
I tried with `htop 3.2.2` and it does as you say: shows bargraphs for all cores regardless of online state, and marks as 'offline' those that are, and updates as soon as the on/offline state changes. Great! Thanks.

**Comment by jorgechamorro** [comment]
Ergo, closing.
